s2a/BUILD.bazel

```diff
@@ -32,33 +32,11 @@ java_library(
 )
 
 java_library(
-    name = "token_fetcher",
-    srcs = ["src/main/java/io/grpc/s2a/internal/handshaker/tokenmanager/TokenFetcher.java"],
+    name = "token_manager",
+    srcs = ["src/main/java/io/grpc/s2a/internal/handshaker/tokenmanager/*.java"],
     deps = [
         ":s2a_identity",
-    ],
-)
-
-java_library(
-    name = "access_token_manager",
-    srcs = [
-        "src/main/java/io/grpc/s2a/internal/handshaker/tokenmanager/AccessTokenManager.java",
-    ],
-    deps = [
-        ":s2a_identity",
-        ":token_fetcher",
         artifact("com.google.code.findbugs:jsr305"),
-    ],
-)
-
-java_library(
-    name = "single_token_fetcher",
```

Comment:
Oh, I understand more of what was going on here, now. You were thinking people didn't have to depend on certain targets. In OSS, they must get all-or-nothing of a particular Java package, and they get everything per-project (aka artifact, aka jar. So grpc-java/s2a folder here).

SingleTokenFetcher has so very, very little code and no dependencies. So it doesn't make sense to split it out into a separate package+project/artifact.